plugins/panels/model_evaluation.py

```diff
@@ -585,6 +610,89 @@ def load_view(self, ctx):
                     view = eval_view.filter_labels(
                         pred_field, F(eval_key) == field, only_matches=True
                     )
+        elif info.config.type == "segmentation":
+            results = ctx.dataset.load_evaluation_results(eval_key)
+            classes_map = self.get_classes_map(ctx.dataset, results, gt_field)
+            if (
+                results.ytrue_ids_dict is None
+                or results.ypred_ids_dict is None
+            ):
+                # legacy segmentation evaluation
+                return
+
+            if eval_key2:
+                if gt_field2 is None:
+                    gt_field2 = gt_field
+
+                results2 = ctx.dataset.load_evaluation_results(eval_key2)
+                classes_map2 = self.get_classes_map(
+                    ctx.dataset, results2, gt_field2
+                )
+                if (
+                    results2.ytrue_ids_dict is None
+                    or results2.ypred_ids_dict is None
+                ):
+                    # legacy segmentation evaluation
+                    return
+            else:
+                results2 = None
+
+            _, gt_id = ctx.dataset._get_label_field_path(gt_field, "_id")
+            _, pred_id = ctx.dataset._get_label_field_path(pred_field, "_id")
+            if gt_field2 is not None:
+                _, gt_id2 = ctx.dataset._get_label_field_path(gt_field2, "_id")
+            if pred_field2 is not None:
+                _, pred_id2 = ctx.dataset._get_label_field_path(
+                    pred_field2, "_id"
+                )
+
+            if view_type == "class":
+                # All GT/predictions that contain class `x`
+                k = classes_map[x]
+                ytrue_ids, ypred_ids = _get_ids_slice(results, k)
+                expr = F(gt_id).is_in(ytrue_ids)
+                expr |= F(pred_id).is_in(ypred_ids)
+                if results2 is not None:
+                    k2 = classes_map2[x]
+                    ytrue_ids2, ypred_ids2 = _get_ids_slice(results2, k2)
+                    expr |= F(gt_id2).is_in(ytrue_ids2)
+                    expr |= F(pred_id2).is_in(ypred_ids2)
+
+                view = eval_view.match(expr)
+            elif view_type == "matrix":
+                # Specific confusion matrix cell
+                i = classes_map[x]
+                j = classes_map[y]
+                ytrue_ids = _to_object_ids(
+                    results.ytrue_ids_dict.get((i, j), [])
+                )
+                ypred_ids = _to_object_ids(
+                    results.ypred_ids_dict.get((i, j), [])
+                )
+                expr = F(gt_id).is_in(ytrue_ids)
+                expr &= F(pred_id).is_in(ypred_ids)
+                view = eval_view.match(expr)
+            elif view_type == "field":
+                if field == "tp":
+                    # All true positives
+                    inds = results.ytrue == results.ypred
```

Comment:
Segmentation tasks do not currently show TP/FP/FN counts in the `Metrics performance section`. But if we were to add them, then this implements the callback for the "grid" icons that would appear.